src/main/java/org/terasology/launcher/gui/javafx/ApplicationController.java

```diff
@@ -199,29 +280,24 @@ protected void openSettingsAction() {
             }
 
             final SettingsController settingsController = fxmlLoader.getController();
-            settingsController.initialize(launcherDirectory, downloadDirectory, launcherSettings, gameVersions, settingsStage);
+            settingsController.initialize(launcherDirectory, downloadDirectory, launcherSettings, packageManager, settingsStage, this);
```

Comment:
I'm not sure about passing `this` here... maybe we can untangle it a bit more later on...

Reply:
The controller classes surely need a good amount of refactoring :wink: